src/carousel/carousel.ts

```diff
@@ -103,14 +105,46 @@ export class NgbCarousel implements AfterContentChecked,
   }
 
   private _cycleToNext() {
-    const nextIdx = this._activeIdx + 1;
-    this.select(nextIdx >= this._slides.length ? (this.wrap ? 0 : nextIdx - 1) : nextIdx);
+    let selectedId: string = this._getNextSlide(this.activeId);
+    this.select(selectedId);
   }
 
   private _restartTimer() {
     this.pause();
     this.cycle();
   }
+
+  private _getSlideById(slideIdx: string): NgbSlide {
+    let slideWithId: NgbSlide[] = this._slides.filter(slide => slide.id === slideIdx);
+    return slideWithId.length ? slideWithId[0] : null;
+  }
+
+  private _getNextSlide(id: string): string {
+    let nextSlideId: string = id;
+    let slideArr: NgbSlide[] = this._slides.toArray();
```

Comment:
Are they needed? `_slides` is typed as `QueryList<NgbSlide>`, and if one looks at the definition of QueryList, it looks like it should work (see https://github.com/angular/angular/blob/master/modules/%40angular/core/src/linker/query_list.ts#L82 and https://github.com/angular/angular/blob/master/modules/%40angular/core/src/linker/query_list.ts#L56-L58). I could be missing something though.